js/src/components/enhanced-conversion-tracking-settings/cta/PendingStatus.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '.~/data';
+import { API_NAMESPACE } from '.~/data/constants';
+import { ENHANCED_ADS_CONVERSION_STATUS } from '.~/constants';
+import usePolling from '.~/hooks/usePolling';
+
+const PendingStatus = () => {
+	const { updateEnhancedAdsConversionStatus, invalidateResolution } =
+		useAppDispatch();
+
+	const { data, start } = usePolling(
+		{
+			path: `${ API_NAMESPACE }/ads/accepted-customer-data-terms`,
+		},
+		20
```

Comment:
Is it possible to stop polling once we know the ToS has been accepted?

Reply:
Yes. I just changed the implementation which works much better.